File: shared/src/main/java/ru/privatenull/pnauth/platform/PlatformScheduler.java
package ru.privatenull.pnauth.platform;

import java.time.Duration;

/** Platform-independent scheduler. Player-bound tasks are safe to use on Folia. */
public interface PlatformScheduler {
    TaskHandle execute(Runnable task);
    TaskHandle execute(PnPlayer player, Runnable task);
    TaskHandle delayed(Duration delay, Runnable task);
    TaskHandle delayed(PnPlayer player, Duration delay, Runnable task);
    TaskHandle repeating(Duration initialDelay, Duration interval, Runnable task);
    TaskHandle repeating(PnPlayer player, Duration initialDelay, Duration interval, Runnable task);
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/platform/PlatformScheduler.kt
package ru.privatenull.pnauth.platform

import java.time.Duration

/** Platform-independent scheduler. Player-bound tasks are safe to use on Folia. */
interface PlatformScheduler {
    fun execute(task: Runnable): TaskHandle
    fun execute(player: Player, task: Runnable): TaskHandle
    fun delayed(delay: Duration, task: Runnable): TaskHandle
    fun delayed(player: Player, delay: Duration, task: Runnable): TaskHandle
    fun repeating(initialDelay: Duration, interval: Duration, task: Runnable): TaskHandle
    fun repeating(player: Player, initialDelay: Duration, interval: Duration, task: Runnable): TaskHandle
}
